src/Microsoft.CodeAnalysis.ResxSourceGenerator/Microsoft.CodeAnalysis.ResxSourceGenerator/AbstractResxGenerator.cs

```diff
@@ -0,0 +1,709 @@
+#pragma warning disable IDE0057 // Use range operator (incorrectly reported when Range is not defined)
+#pragma warning disable IDE0058 // Expression value is never used (not sure why this is enabled)
+#pragma warning disable IDE0066 // Convert switch statement to expression (not always better)
+
+namespace Microsoft.CodeAnalysis.ResxSourceGenerator
+{
+    internal abstract class AbstractResxGenerator : ISourceGenerator
+    {
+        protected abstract bool SupportsNullable(GeneratorExecutionContext context);
+
+        public void Initialize(GeneratorInitializationContext context)
+        {
+        }
+
+        [SuppressMessage("Design", "CA1031:Do not catch general exception types", Justification = "Standard practice for diagnosing source generator failures.")]
+        public void Execute(GeneratorExecutionContext context)
+        {
+            var resourceFiles = context.AdditionalFiles.Where(file => file.Path.EndsWith(".resx", StringComparison.OrdinalIgnoreCase));
+            foreach (var resourceFile in resourceFiles)
+            {
+                try
+                {
+                    ProcessResourceFile(context, resourceFile);
+                }
+                catch (Exception ex)
+                {
+                    var exceptionLines = ex.ToString().Split(new[] { Environment.NewLine }, StringSplitOptions.None);
+                    var text = string.Join("", exceptionLines.Select(line => "#error " + line + Environment.NewLine));
+                    var errorText = SourceText.From(text, Encoding.UTF8, SourceHashAlgorithm.Sha256);
+                    context.AddSource($"{Path.GetFileName(resourceFile.Path)}.Error", errorText);
+                }
+            }
+        }
+
+        private void ProcessResourceFile(GeneratorExecutionContext context, AdditionalText resourceFile)
+        {
+            var options = context.AnalyzerConfigOptions.GetOptions(resourceFile);
+
+            if (options.TryGetValue("build_metadata.AdditionalFiles.GenerateSource", out var generateSourceText)
+                && !string.IsNullOrEmpty(generateSourceText)
+                && generateSourceText != "unset"
+                && !string.Equals(bool.TrueString, generateSourceText, StringComparison.OrdinalIgnoreCase))
+            {
+                // Source generation is disabled for this resource file
+                return;
+            }
+
+            if (!context.AnalyzerConfigOptions.GlobalOptions.TryGetValue("build_property.RootNamespace", out var rootNamespace))
+            {
+                rootNamespace = context.Compilation.AssemblyName;
+            }
+
+            var resourceName = Path.GetFileNameWithoutExtension(resourceFile.Path);
+            if (options.TryGetValue("build_metadata.AdditionalFiles.RelativeDir", out var relativeDir))
+            {
+                resourceName = relativeDir.Replace(Path.DirectorySeparatorChar, '.').Replace(Path.AltDirectorySeparatorChar, '.') + resourceName;
+            }
```

Comment:
IIRC the name of the type is actually more nuanced than this.  I believe this is the default but other things can influence the actual value.  Doesn't the resx item itself have this on it?

Reply:
> Doesn't the resx item itself have this on it?

I was going based on what I saw in Arcade. If there's a better way to achieve this let me know.

Reply:
Did you check out the targets there and in MSBuild? https://github.com/dotnet/arcade/blob/main/src/Microsoft.DotNet.Arcade.Sdk/tools/GenerateResxSource.targets#L75

I don't imagine this source generator is going to have / require targets?